src/DotenvAdapter.php

```diff
@@ -48,15 +49,20 @@ public function exposeToEnvironment(string $envFile): void
                 $envFile = $envDir . DIRECTORY_SEPARATOR . $envFile;
             }
 
-            if (!is_readable($envFile)) {
+            if (!is_readable($envFile) || is_dir($envFile)) {
                 continue;
             }
 
-            $dotEnv->load($envFile);
+            foreach ($dotEnv->parse(file_get_contents($envFile), $envFile) as $var => $val) {
+                if ($_SERVER[$var] ?? '') {
+                    $val = $_SERVER[$var];
+                }
 
-            // Reset Symfony loaded dotenv vars
-            // We need to do this or Symfony will override previously set dotenv vars
-            $_SERVER['SYMFONY_DOTENV_VARS'] = $_ENV['SYMFONY_DOTENV_VARS'] = '';
```

Comment:
Because we're now using `$dotEnv->parse()` directly and do the populate-part ourselves, instead of using `$dotEnv->load()` which calls `$dotEnv->parse()` and `$dotEnv->populate()`.